pkg/generate/code/check_test.go

```diff
@@ -115,3 +115,24 @@ func TestCheckRequiredFields_RenamedSpecField(t *testing.T) {
 		strings.TrimSpace(gotCode),
 	)
 }
+
```

Comment:
I wanted a test where the Shape/CR have multiple identifiers, but the `Id` gets prioritized. The only one I could think of was *SecurityGroups*. The DescribeInput can take either *Name* or *Id*, and the SecurityGroup CR has both Name in the Spec and Id in the Status. 

In this scenario `Id` should be a required field. However, when trying to test this I uncovered an edge case in the `FindIdentifiers` helper funcs. No identifiers are returned for SecurityGroup CR or the DescribeInput because the fields are not securityGroupName/securityGroupId, but groupName/groupId. 😒

I can address this edge case in a different PR, but if anyone else has a good API to test the above case I'd love to add and test it now!